File: src/lib/server/validation/responses/account.ts
import { z } from 'zod';

export const accountResponseSchema = z
	.object({
		user: z
			.object({
				email: z.string().nullable(),
				name: z.string().nullable(),
				createdAt: z.string().nullable()
			})
			.meta({ id: 'AccountUser' }),
		dataRange: z
			.object({
				earliest: z.string().nullable(),
				latest: z.string().nullable()
			})
			.meta({ id: 'AccountDataRange' })
	})
	.meta({ id: 'AccountResponse' });

File: mobile/iosApp/PhoneShared/Models/Account.swift
import Foundation

struct AccountUser: Codable {
    let email: String?
    let name: String?
    let createdAt: String?
}

/// Earliest/latest dated row in the account. The download bounds itself with
/// this rather than with `createdAt`: entry, sleep, weight and day-property
/// dates are client-chosen, so imported or backfilled days can predate the
/// account and a device ahead of UTC can log past the server's "today".
struct AccountDataRange: Codable {
    let earliest: String?
    let latest: String?
}

struct AccountResponse: Codable {
    let user: AccountUser
    let dataRange: AccountDataRange
}
